src/platform/forms-system/src/js/utilities/data/reduceErrors.js

```diff
@@ -0,0 +1,223 @@
+ * @property {string} pageKey - the page within the chapter the element is
+ *   associated with
+ */
+/**
+ * @typedef Form~rawErrors - list of raw errors that are output from jsonschema
+ *   validator
+ * @type {object}
+ * @property {string} property - path within the page uiSchema that leds to the
+ *   error. Always starts with "instance"
+ * @property {string} message - specific error message. It may or may not
+ *   include the property causing the error
+ * @property {string} name - When provided, it includes the type of error, e.g.
+ *   "required", "minItems", "maxItems", etc
+ * @property {string|number} argument - name of the property or index in array
+ *   with the validation error
+ * @property {object} schema - schema of the page; ignored by this error reducer
+ * @property {string} stack - error message used in processing; similar to the
+ *   error message, but includes the path to the error
+ * @property {string[]} __errors - may contain multiple error messages for one
+ *   element
+ *
+ * There are five types of hardcoded validation error messages:
+ *
+ * min/max: {
+    property: 'instance.someProperty',
+    message: 'does not meet minimum length of 1',
+    name: 'minItems',
+    argument: 1,
+    schema: {...},
+    stack: 'instance.someProperty does not meet minimum length of 1'
+  }
+  * required: {
+    property: 'instance',
+    message: 'requires property "someProperty"',
+    name: 'required',
+    argument: 'someProperty',
+    schema: {...},
+    stack: 'instance requires property "someProperty"'
+  }
+  * array: {
+    property: 'instance.newDisabilities[0]',
+    message: 'requires property "cause"',
+    name: 'required',
+    argument: 'cause',
+    schema: {...},
+    stack: 'instance.newDisabilities[0] requires property "cause"'
+  }
+  * array w/deeply nested property: {
+    property: 'instance.newDisabilities[4].view:secondaryFollowUp.causedByDisability',
+    message: 'is not one of enum values: Abc,Bcd,Cde,Def,Efg,Fgh,Ghi,Hij',
+    schema: {...},
+  }
+  * array with argument array: {
+    argument: ['Abc', 'Bcd', 'Cde', 'Def', 'Efg', 'Fgh', 'Ghi', 'Hij'],
+    message: 'is not one of enum values: Abc,Bcd,Cde,Def,Efg,Fgh,Ghi,Hij',
+    name: 'enum',
+    property: 'instance.newDisabilities[6].view:secondaryFollowUp.causedByDisability',
+    schema: {},
+    stack: 'instance.newDisabilities[6].view:secondaryFollowUp.causedByDisability is not one of enum values: Abc,Bcd,Cde,Def,Efg,Fgh,Ghi,Hij',
+  }
+  * non-empty "__errors" array: {
+    __errors: ['error message']
+  }
+*/
+/**
+ * @typedef Form~formErrors
+ * @type {object}
+ * @property {Form~rawErrors}
+ * @property {Form~errors}
+ */
+/**
+ * Process rawErrors from jsonschema validator into a more friendly list
+ * @param {Form~formErrors} errors
+ * @param {Form~pageList} pageList - list of all form pages from `route.pageList`
+ */
+export const reduceErrors = (errors, pageList) =>
+  errors.reduce((result, error) => {
```

Comment:
So I haven't looked at this code in a loooonng time... `reduceErrors` has raw errors & the complete `pageList` as parameters and returns a reduced list of form errors

```js
/**
 * Process rawErrors from jsonschema validator into a more friendly list
 * @param {Form~rawErrors} errors - raw form errors from jsonschema validator
 * @param {Form~pageList} pageList - list of all form pages from `route.pageList`
 * @return {Form~errors} - Finely curated list of form errors
 */
```

I think I'll remove the `Form~formErrors` JSDoc type definition

```js
/**
 * @typedef Form~formErrors - Resulting Redux `form.formErrors` object created
 *   within the SubmitController
 * @type {object}
 * @property {Form~rawErrors} - raw form errors produced by jsonschema validator
 * @property {Form~errors} - cleaned up form errors produced by the reduceErrors
 *   function
 */
```

because it's really only in the format in the Redux state (set by the `SubmitController`)

----

As for the `reduce` function parameter, what about `processedErrors`? That's the variable name of this function returned value in the `SubmitController`